fe/fe-core/src/main/java/org/apache/doris/analysis/ShowPartitionsStmt.java

```diff
@@ -137,8 +151,14 @@ public void analyze(Analyzer analyzer) throws UserException {
     public void analyzeImpl(Analyzer analyzer) throws UserException {
         super.analyze(analyzer);
         tableName.analyze(analyzer);
-        // disallow external catalog
-        Util.prohibitExternalCatalog(tableName.getCtl(), this.getClass().getSimpleName());
+        // disallow unsupported catalog
+        String catalogName = tableName.getCtl();
+        catalog = Env.getCurrentEnv().getCatalogMgr().getCatalog(catalogName);
+        String catalogType = catalog.getType();
+        if (!Strings.isNullOrEmpty(catalogName) && !(catalog.isInternalCatalog() || catalogType.equals("hms"))) {
```

Comment:
No need to check `!Strings.isNullOrEmpty(catalogName)`, because after `tableName.analyze(analyzer)`, the catalogName will be filled.
And you can use `catalog instanceof xxx` to check catalog's type, instead of using string compare.

Reply:
fixed